From stern@rowland.harvard.edu  Tue Nov 25 13:45:29 2008
From: Alan Stern <stern@rowland.harvard.edu>
Date: Tue, 25 Nov 2008 16:40:02 -0500 (EST)
Subject: USB: utilize round_jiffies_up_relative()
To: Greg KH <greg@kroah.com>
Message-ID: <Pine.LNX.4.44L0.0811251639190.3526-100000@iolanthe.rowland.org>


This patch (as1178) uses the new round_jiffies_up_relative() routine
for setting the autosuspend delayed_work timer.  It's appropriate
since we don't care too much about the exact length of the delay, but
we don't want it to be too short (rounded down).

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/driver.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -1111,7 +1111,7 @@ static int autosuspend_check(struct usb_
 	if (reschedule) {
 		if (!timer_pending(&udev->autosuspend.timer)) {
 			queue_delayed_work(ksuspend_usb_wq, &udev->autosuspend,
-				round_jiffies_relative(suspend_time - j));
+				round_jiffies_up_relative(suspend_time - j));
 		}
 		return -EAGAIN;
 	}
@@ -1553,7 +1553,7 @@ void usb_autopm_put_interface_async(stru
 		else if (intf->pm_usage_cnt <= 0 &&
 				!timer_pending(&udev->autosuspend.timer)) {
 			queue_delayed_work(ksuspend_usb_wq, &udev->autosuspend,
-					round_jiffies_relative(
+					round_jiffies_up_relative(
 						udev->autosuspend_delay));
 		}
 	}
